actioncable/lib/action_cable/connection/base.rb

```diff
@@ -52,47 +51,37 @@ class Base
       include Callbacks
       include ActiveSupport::Rescuable
 
-      attr_reader :server, :env, :subscriptions, :logger, :worker_pool, :protocol
-      delegate :event_loop, :pubsub, :config, to: :server
+      attr_reader :subscriptions, :logger
+      private attr_reader :server, :raw_conn
 
-      def initialize(server, env, coder: ActiveSupport::JSON)
-        @server, @env, @coder = server, env, coder
+      delegate :pubsub, :config, to: :server
+      delegate :env, :request, :protocol, :perform_work, to: :raw_conn, allow_nil: true
 
-        @worker_pool = server.worker_pool
-        @logger = new_tagged_logger
+      def initialize(server, raw_conn)
+        @server = server
+        @raw_conn = raw_conn
 
-        @websocket      = ActionCable::Connection::WebSocket.new(env, self, event_loop)
-        @subscriptions  = ActionCable::Connection::Subscriptions.new(self)
-        @message_buffer = ActionCable::Connection::MessageBuffer.new(self)
+        @logger = raw_conn.logger
+        @subscriptions  = Subscriptions.new(self)
 
         @_internal_subscriptions = nil
+
         @started_at = Time.now
       end
 
-      # Called by the server when a new WebSocket connection is established. This configures the callbacks intended for overwriting by the user.
-      # This method should not be called directly -- instead rely upon on the #connect (and #disconnect) callbacks.
-      def process # :nodoc:
-        logger.info started_request_message
-
-        if websocket.possible? && allow_request_origin?
-          respond_to_successful_request
-        else
-          respond_to_invalid_request
-        end
+      def handle_open
+        connect if respond_to?(:connect)
```

Comment:
What about just having an empty connect method rather than using `if respond_to?`

Reply:
Yeah, that's a good question. I ported it from the current version; I will try to dig why it has been implemented this way in the first place.

Reply:
Done!